torch_xla/csrc/reduction.cpp

```diff
@@ -422,6 +422,24 @@ xla::XlaOp BuildAll(xla::XlaOp input, absl::Span<const xla::int64> dimensions,
   xla::XlaOp result =
       xla::Reduce(input, init_value, CreateAllComputation(shape.element_type()),
                   dimensions);
+  bool all_reduce_dims_size_one = true;
+  for (xla::int64 dim : dimensions) {
+    if (shape.dimensions(dim) != 1) {
+      all_reduce_dims_size_one = false;
+      break;
+    }
+  }
+  if (all_reduce_dims_size_one) {
+    const xla::Shape& result_shape = XlaHelpers::ShapeOfXlaOp(result);
+    xla::XlaOp zeros =
+        XlaHelpers::ScalarBroadcast<float>(0.0, result_shape, input.builder());
+    xla::XlaOp ones =
+        XlaHelpers::ScalarBroadcast<float>(1.0, result_shape, input.builder());
+    // This is to force the result value to be 1 or 0 in the case of all reduce
+    // dimensions have size one. Reduce won't perform AllComputation in this
+    // case.
+    result = xla::Select(xla::Ne(result, zeros), ones, zeros);
```

Comment:
I'd check with Blake to see what's faster on TPU. The above or just use `MaybeConvertTo()`.

Reply:
I checked with Blake and these two approach will have the same performance. I switched to `MaybeConvertTo` since the code is cleaner.